pkg/util/watch/relist.go

```diff
@@ -20,14 +21,33 @@ type relistInfo struct {
 	//
 	// Refer to the usage in (*relistExecutor).CollectTriggeredRequests() for more.
 	relisted chan struct{}
+
+	// relistCount is the number of instances of relisted that have been closed.
+	//
+	// Externally, we only guarantee that this is monotonically increasing; internally, it's
+	// associated each time with a single instance of `relisted`, so that RelistIfHaventSince can
+	// hook into ongoing execution.
+	relistCount RelistCount
+
+	// executor stores a reference to the current relistExecutor, if there is one active.
+	executor *relistExecutor
 }
 
+// RelistCount is a monotonically increasing value to track (approximately) the number of relists
```

Comment:
Technically, this currently represents the number of calls to `List()`, but "number of relists" might be assumed to be the entire relist process, ignoring the number of failed requests.

I wasn't quite sure how to put that, and also don't want to make that part of the public API — really, I'd rather the guarantees of the API be more like "this is a monotonically increasing value that increases when there's a relist but _may_ increase otherwise; don't assume anything beyond that". But... who is this API for, y'know? Feels like it doesn't make as much sense to be so strict when it's not _that_ big, and just for internal use.